src/compiler/decorator/Decorator.ts

```diff
@@ -274,4 +277,11 @@ export class Decorator extends DecoratorBase<ts.Decorator> {
                 getSiblingFormatting: (parent, sibling) => sibling.getStartLinePos() === thisStartLinePos ? FormattingKind.Space : FormattingKind.Newline
             });
     }
+
+    getStructure(): DecoratorStructure {
+        return {
+            name: this.getName(),
+            arguments: this.getArguments().map(arg => arg.getText())
```

Comment:
Careful here. Providing an empty array will make it a decorator factory (ex `@decorator()`). This needs to be undefined when there's no parenthesis (ex. `@decorator`)